### PR TITLE
(MODULES-4314) Updates to get CI passing

### PR DESCRIPTION
**Pull request by johnduarte**

### CODE REVIEW

**Comment by johnduarte** [comment]
DO NOT MERGE
This PR fails to address permissions issues on the puppet labs jenkins runner that cause the spec test job to fail.
```
     RuntimeError:
       Got 2 failure(s) while initializing: File[/tmp/rspecrun20170309-30029-1kqof18/var20170309-30029-144h6yl]: change from jenkins to puppet failed: Failed to set group to '111': Operation not permitted @ chown_internal - /tmp/rspecrun20170309-30029-1kqof18/var20170309-30029-144h6yl; File[/tmp/rspecrun20170309-30029-1kqof18/conf20170309-30029-15np99l/ssl]: change from absent to directory failed: Failed to set group to '111': Operation not permitted @ chown_internal - /tmp/rspecrun20170309-30029-1kqof18/conf20170309-30029-15np99l/ssl
     # ./.bundle/gems/gems/puppet-4.7.0/lib/puppet/settings.rb:1011:in `block in use'
     # ./.bundle/gems/gems/puppet-4.7.0/lib/puppet/resource/catalog.rb:230:in `apply'
     # ./.bundle/gems/gems/puppet-4.7.0/lib/puppet/settings.rb:1001:in `use'
     # ./.bundle/gems/gems/puppet-4.7.0/lib/puppet/ssl/certificate_authority.rb:156:in `initialize'
     # ./spec/spec_helper_local.rb:40:in `new'
     # ./spec/spec_helper_local.rb:40:in `generate_pki'
     # ./spec/spec_helper_local.rb:50:in `block (2 levels) in <top (required)>'
```

**Comment by adrienthebo** [comment]
@johnduarte if you need PRs against this repo to be merged and I don't respond on my own, feel free to directly ping me about reviewing/merging PRs.

**Comment by johnduarte** [comment]
The root cause of the permissions issue on the Jenkins runner has been identified. This is caused by the puppetlabs_spec_helper injecting a stub that causes puppet to believe it is always running as root.
https://github.com/puppetlabs/puppetlabs_spec_helper/blob/master/lib/puppetlabs_spec_helper/module_spec_helper.rb#L29
This leads to attempts to execute `chown` in a manner that the user running the spec tests does not have permission to do.
This PR includes a commit to unstub `Puppet.features.root?` in order for the puppet code to be executed in accordance with the user that is running it. This change allows the spec tests to pass in CI.

@adrienthebo and @glennsarti please chime in if you have concerns about this.

**Comment by glennsarti** [comment]
Thanks John.  We're just reviewing the puppetlabs-spec-helper to see if that stub should even be there too.

File: metadata.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "version": "0.1.1",
   "author": "puppetlabs",
   "summary": "Puppet module providing the certregen face for regenerating CA certificates",
-  "license": "Puppet Enterprise Software License Agreement",
+  "license": "proprietary",
```
Comment (adrienthebo) [inline metadata.json]: This conflicts with the contents of the `NOTICE` file that indicates that the module is Apache licensed.

**Comment by adrienthebo** [comment]
1 comment, LGTM otherwise.

File: metadata.json (hunk 1 above)
Comment (johnduarte) [inline metadata.json]: Is 'Apache' the correct license for this module?

Comment (johnduarte) [inline metadata.json]: I assume the answer to this question is no given the previous value here. I have updated the PR with the followiing changes:
1. The NOTICE file has been deleted
2. The LICENSE file has been replaced with the `Puppet Enterprise Software License Agreement`
These changes bring the licensing in parity with the [sqlserver module](https://github.com/puppetlabs/puppetlabs-sqlserver)

Comment (glennsarti) [inline metadata.json]: Probably best to use the websphere module as the SQL Server one is wrong.

Comment (johnduarte) [inline metadata.json]: So, this module should use the 'Apache-2.0' license?

**Comment by adrienthebo** [comment]
@johnduarte let's deal with licensing in another PR; for the sake of CI I've merged this.

**Pull request merged**